android-json-form-wizard/src/main/java/com/vijay/jsonwizard/task/AttachRepeatingGroupTask.java

```diff
@@ -130,13 +134,51 @@ protected void onPostExecute(List<View> result) {
         }
 
         try {
-            ((JsonApi) widgetArgs.getContext()).invokeRefreshLogic(null, false, null, null);
+            ((JsonApi) widgetArgs.getContext()).invokeRefreshLogic(null, false, null, null, widgetArgs.getStepName(), false);
         } catch (Exception e) {
             Timber.e(e);
         }
 
         hideProgressDialog();
         doneButton.setImageResource(R.drawable.ic_done_green);
+
+        //for validation
+        validationCleanUp();
+
+    }
+
+    private void validationCleanUp() {
+        doneButton.setTag(R.id.is_repeating_group_generated, true);
+
+        LinearLayout referenceLayout = (LinearLayout) ((LinearLayout) parent).getChildAt(0);
+        MaterialEditText materialEditText = (MaterialEditText) referenceLayout.getChildAt(0);
+        materialEditText.setError(null);
+
+        widgetArgs.getFormFragment().getJsonApi().getAppExecutors().diskIO().execute(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    JSONObject step = ((JsonApi) widgetArgs.getContext()).getmJSONObject().getJSONObject(widgetArgs.getStepName());
+                    JSONArray fields = step.getJSONArray(FIELDS);
+                    String key = widgetArgs.getJsonObject().optString(KEY);
+                    for (int i = 0; i < fields.length(); i++) {
+                        JSONObject object = fields.optJSONObject(i);
+                        String fieldKey = object.optString(KEY);
+                        if (fieldKey.equals(key)) {
+                            String strNumOfRepeatedGroups = object.optString(JsonFormConstants.REPEATING_GROUPS_GENERATED_COUNT);
+                            int currentCount = numRepeatingGroups;
+                            if (StringUtils.isNotBlank(strNumOfRepeatedGroups)) {
+                                currentCount += Integer.parseInt(strNumOfRepeatedGroups);
+                            }
+                            object.put(JsonFormConstants.REPEATING_GROUPS_GENERATED_COUNT, currentCount);
+                            break;
+                        }
+                    }
+                } catch (JSONException e) {
+                    Timber.e(e);
+                }
```

Comment:
This validation logic may not be necessary after this merges https://github.com/OpenSRP/opensrp-client-native-form/pull/476.

This count can be retrieved from the count object.

Reply:
It is since #476 does not ensure the repeating group is actually created ... it only checks for text change

Reply:
It can be modified to only write into the count object after repeating group generation. And that can be used to verify the generation was done.

Reply:
will update this

Reply:
https://github.com/OpenSRP/opensrp-client-native-form/pull/504